src/test/java/io/confluent/connect/elasticsearch/ElasticsearchClientTest.java

```diff
@@ -0,0 +1,451 @@
+import io.confluent.connect.elasticsearch.helper.ElasticsearchHelperClient;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.sink.ErrantRecordReporter;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.test.TestUtils;
+import org.elasticsearch.ElasticsearchStatusException;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.search.SearchHit;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.testcontainers.shaded.okio.Sink;
+
+public class ElasticsearchClientTest {
+
+  private static final String INDEX = "index";
+
+  private static ElasticsearchContainer container;
+
+  private DataConverter converter;
+  private ElasticsearchHelperClient helperClient;
+  private ElasticsearchSinkConnectorConfig config;
+  private Map<String, String> props;
+
+  @BeforeClass
+  public static void setupBeforeAll() {
+    container = ElasticsearchContainer.fromSystemProperties();
+    container.start();
+  }
+
+  @AfterClass
+  public static void cleanuoAfterAll() {
+    container.close();
+  }
+
+  @Before
+  public void setup() {
+    props = ElasticsearchSinkConnectorConfigTest.addNecessaryProps(new HashMap<>());
+    props.put(CONNECTION_URL_CONFIG, container.getConnectionUrl());
+    props.put(IGNORE_KEY_CONFIG, "true");
+    config = new ElasticsearchSinkConnectorConfig(props);
+    converter = new DataConverter(config);
+    helperClient = new ElasticsearchHelperClient(container.getConnectionUrl());
+  }
+
+  @After
+  public void cleanup() throws IOException {
+    if (helperClient.indexExists(INDEX)){
+      helperClient.deleteIndex(INDEX);
+    }
+  }
+
+  @Test
+  public void testClose() {
+    ElasticsearchClient client = new ElasticsearchClient(config, null);
+    client.close();
+  }
+
+  @Test
+  public void testCreateIndex() throws IOException {
+    ElasticsearchClient client = new ElasticsearchClient(config, null);
+    assertFalse(helperClient.indexExists(INDEX));
+
+    client.createIndex(INDEX);
+    assertTrue(helperClient.indexExists(INDEX));
+  }
+
+  @Test
+  public void testIndexAlreadyExists() throws IOException {
+    ElasticsearchClient client = new ElasticsearchClient(config, null);
+    assertFalse(helperClient.indexExists(INDEX));
+
+    client.createIndex(INDEX);
+    assertTrue(helperClient.indexExists(INDEX));
+
+    client.createIndex(INDEX);
+    assertTrue(helperClient.indexExists(INDEX));
+  }
+
+  @Test
+  public void testIndexExists() throws IOException {
```

Comment:
What is the difference between this one and `testIndexAlreadyExists`?

Reply:
`testIndexAlreadyExists` tests that the client does not try to create an index that does not already exist

I will update the name